### ISSUE TITLE
'main' field in package.json

### ISSUE THREAD

**Issue by fritx** [FortAwesome/Font-Awesome#8368]
See: Make normalize.css available for webpack
https://github.com/necolas/normalize.css/pull/421

**Comment by tagliala** [comment]
Sorry, I didn't get this request

Could you please clarify?

**Comment by fritx** [comment]
OK. I meant the `main` field is missing in package.json.
I can't `import` it with webpack. The issue is similar with necolas/normalize.css#421.
Any pull request is welcomed?

**Comment by tagliala** [comment]
Sorry but I'm not so much into node and npm

I'm noticing that `main` seems related to `.js` files

https://docs.npmjs.com/files/package.json#main

> The main field is a module ID that is the primary entry point to your program. That is, if your package is named foo, and a user installs it, and then does require("foo"), then your main module's exports object will be returned.

> This should be a module ID relative to the root of your package folder.

> For most modules, it makes the most sense to have a main script and often not much else.

EG: bootstrap https://github.com/twbs/bootstrap/blob/master/package.json#L22

since there is no javascript here, what is the use case of pointing main to a css file?

**Comment by fritx** [comment]
hi @tagliala
npm is not only for js now. many css and other modules are available on it, such as normalize.css, 
and we prefer to require them in js thanks to webpack bundling.

**Comment by tagliala** [comment]
hi @fritx,

thanks for your clarification. I need feedback from Dave before adding this

**Comment by fritx** [comment]
cool awesome :)

**Comment by romiem** [comment]
+1 Please add a main property in the package.json file which points to ./css/font-awesome.css'.

**Comment by Youmoo** [comment]
+1 [bulma](https://github.com/jgthms/bulma/blob/master/package.json#L7) also does this.

**Comment by tothandras** [comment]
@tagliala sorry for including you but I see this is open for a long time and it would be a quick fix :) I am happy to open a PR too if it makes easier.

**Comment by tagliala** [comment]
@tothandras sorry I can't make this decision, please check if you can feedback from him

**Comment by niftylettuce** [comment]
Please add this!

**Comment by jrharshath** [comment]
Update/FYI: Bootstrap has not only JS, but also CSS. They're using a "style" field in package.json for pointing to css (or scss, in case of bootstrap-sass), and the "main" field for their JS.

**Comment by fritx** [comment]
@jrharshath is this issue related? https://github.com/webpack/css-loader/issues/51
If using the style field, i'm not sure, does `import 'font-awesome'` work? 😕

**Comment by jrharshath** [comment]
I'm not using webpack myself - I'm using `node-sass` + `node-sass-import`, and `node-sass-import` makes uses the style fields when I use `@import 'font-awesome'` or `@import 'bootstrap-sass'`.

I think as more and more projects start adopting the `style` or `css` field in their package.json, npm will address it and standardize this to one prescribed convention.

**Comment by bjankord** [comment]
+1 for `style` field

**Comment by oluckyman** [comment]
So guys, it’s one line of code, it have no side-effects and it unlocks all webpack users which are legions out there.  What are you waiting for? Pull request or something? Let’s close that issue and move on

**Comment by oluckyman** [comment]
Meanwhile made a workaround for webpack:
```lang=js
var config = {
  entry: {
    app: '...'
    vendor: getVendors(),
  },
  ...
};

function getVendors() {
  var dependencies = require('./package.json').dependencies || {};
  var vendors = Object.keys(dependencies).map(vendor => {
    switch (vendor) {
      case 'font-awesome':
        return 'font-awesome/css/font-awesome.css';
      default:
        return vendor;
    }
  });
  return vendors;
}
```

**Comment by mikpetr** [comment]
Hi all.
Totally agree with @oluckyman . It's already 2017 and it's not fixed yet. All npm packages have main field. There is no difference which kind of file it is.

**Comment by CyberCookie** [comment]
@davegandy 
Lol really !? Why are you so slow?
This enchancment not only gives benefits for webpack users, but it also allows to retrieve absolute pass to file specified in package.json main by NodeJS **require.resolve('font-awesome')** method. This can be used to read any file font-awesome module consists of.
Cmon ! How much more PRs should be created before you finally merge this 1 line of text !?

**Comment by mikpetr** [comment]
Seems font awesome no more supported.

**Comment by franciscop** [comment]
I added more explanations in the PR I made (above), not only people going through Webpack but also people going through Create-React-App* are affected. With this change implemented, this could be done:

```js
import React from 'react';
import ReactDOM from 'react-dom';
import 'picnic';
import 'font-awesome';

// ...
```

From reading this discussion, it seems that this is the current status:

Pros:

- CDNs get a lot easier to use, from jsDelivery to unpkg.
- Webpack and by extension Create-React-App users lives are a lot simpler.
- Follows the convention used by many other CSS libraries.
- It is retrocompatible. It seems like nothing would break adding this.

Cons:

- It is not what the `main` is designed for.

*While CRA uses webpack in the back, I separate these users for two reasons: they are more likely to be new and not know how to manage these things, and because the examples in CRA are all doing `import './NAME.css';` so I think it is fair to say that they expect this functionality by default.
